src/render.rs

```diff
@@ -0,0 +1,140 @@
+use matrix_sdk::events::room::{
+    encrypted::EncryptedEvent,
+    member::{MemberEvent, MembershipState},
+    message::{
+        AudioMessageEventContent, EmoteMessageEventContent,
+        FileMessageEventContent, ImageMessageEventContent, MessageEvent,
+        MessageEventContent, NoticeMessageEventContent,
+        TextMessageEventContent, VideoMessageEventContent,
+    },
+};
+
+/// This trait describes events that can be rendered in the weechat UI
+pub(crate) trait RenderableEvent {
+    /// Convert the event into a string that will be displayed in the UI.
+    /// The displayname is taken as a parameter since it cannot be calculated from the event
+    /// context alone.
+    fn render(&self, displayname: &str) -> String;
+}
+
+impl RenderableEvent for EncryptedEvent {
+    // TODO: this is not implemented yet
+    fn render(&self, displayname: &str) -> String {
+        format!("{}\t{}", displayname, "Unable to decrypt message")
+    }
+}
+
+impl RenderableEvent for MemberEvent {
+    fn render(&self, displayname: &str) -> String {
+        let operation = match self.content.membership {
+            MembershipState::Join => "joined",
+            MembershipState::Leave => "left",
+            MembershipState::Ban => "banned",
+            MembershipState::Invite => "invited",
+            MembershipState::Knock => "knocked on", // TODO
+        };
+        format!(
+            "{} ({}) has {} the room",
```

Comment:
We'll want to put some color into these, e.g. the delimiters. Another thing those messages have a prefix usually.

You can use `Weechat::color("chat_delimiters")` and `Weechat::prefix("join")` to get them.

Reply:
I'm new to writing weechat plugins. Both `color` and `prefix` return a `&str` but how do I use the return value? Is it used like [this](https://github.com/poljar/weechat-matrix-rs/pull/11/commits/519bbd2561c986702861a369b347bfe026786486)?

Reply:
Correct, you just join it into the final string. More examples of this can be found in the [Python script repo](https://github.com/poljar/weechat-matrix/blob/master/matrix/buffer.py#L721).

Reply:
I inserted some colors, but I'm not sure which prefix to use for regular messages.

Also currently we don't have per-user colors yet, so I just use the same color for each user

Reply:
Regular messages don't have a prefix, at least not one that you'll fetch with `Weechat::prefix()`.

Colors for nicks can be fetched with `Weechat::info_get("nick_color_name", user_id)`, at least I think we want to use the user_id, so the color doesn't change when the display name changes.

Those will need to be stored somewhere like it's done [here](https://github.com/poljar/weechat-matrix/blob/master/matrix/buffer.py#L152).

But we can leave nick colors for another PR. They aren't required to get this merged.